src/components/FormatText.js

```diff
@@ -0,0 +1,88 @@
+/* @flow */
+import React from "react"
+import type {Node as ReactNode, AbstractComponent} from "react"
+
+import Link from "./base/Link"
+
+type Props = {
+    children: ReactNode,
+    paragraphWrapperElement?: string | AbstractComponent<{}, mixed>
+}
+
+function FormatText({children, paragraphWrapperElement: ParagraphWrapper = "p"}: Props): ReactNode {
```

Comment:
I feel this component needs a better docstring to make its purpose clearer as regexes are not really intuitive unless you fancy yourself a wizard.